tempodb/blocklist/poller.go

```diff
@@ -85,13 +121,37 @@ func (p *Poller) Do() (PerTenant, PerTenantCompacted, error) {
 }
 
 func (p *Poller) pollTenant(ctx context.Context, tenantID string) ([]*backend.BlockMeta, []*backend.CompactedBlockMeta, error) {
+	// if we're not building the tenant index then attempt to pull it
+	if !p.buildTenantIndex() {
+		metricTenantIndexBuilder.Set(0)
+
+		i, err := p.reader.TenantIndex(ctx, tenantID)
+		if err == nil {
+			metricTenantIndexAgeSeconds.WithLabelValues(tenantID).Set(float64(time.Since(i.CreatedAt) / time.Second))
+			level.Info(p.logger).Log("msg", "successfully pulled tenant index", "tenant", tenantID, "createdAt", i.CreatedAt, "metas", len(i.Meta), "compactedMetas", len(i.CompactedMeta))
+			return i.Meta, i.CompactedMeta, nil
+		}
+
+		metricTenantIndexErrors.WithLabelValues(tenantID).Inc()
+
+		// we had an error, do we bail?
```

Comment:
What happens if compactors are rolled? That could now impact the state of the blocklist. As a safety catch, should we also fallback to polling if the tenantIndex is older than `2 * BlocklistPoll`?

Reply:
For the moment I would prefer responding to the alert/metric `tempodb_blocklist_tenant_index_age_seconds` and treating it like an operational issue.

I expect this will not happen often, but if it does we can add an option in a future PR to fallback if the index age > some value.